code/src/sixsq/nuvla/server/resources/infrastructure_service_swarm.clj

```diff
@@ -44,61 +43,53 @@
-;                        (throw (r/ex-response "unable to create async job to start deployment" 500 id)))
-;                  (-> id
-;                      (db/retrieve request)
-;                      (a/can-modify? request)
-;                      (assoc :state "STARTING")
-;                      (db/edit request))
-;                  (event-utils/create-event id job-msg (acl/default-acl (acl/current-authentication request)))
-;                  (r/map-response job-msg 202 id job-id))
-;             (catch Exception e
-;               (or (ex-data e) (throw e)))))
\ No newline at end of file
+  (try
+    (let [id (:id service)
+          user-id (:identity (a/current-authentication request))
+          {{job-id     :resource-id
+            job-status :status} :body} (job/create-job id "start_infrastructure_service_swarm"
+                                                       {:owner {:principal "ADMIN"
+                                                                :type      "ROLE"}
+                                                        :rules [{:principal user-id
+                                                                 :right     "VIEW"
+                                                                 :type      "USER"}]}
+                                                       :priority 50)
+          job-msg (str "starting " id " with async " job-id)]
+      (when (not= job-status 201)
+        (throw (r/ex-response "unable to create async job to start infrastructure service swarm" 500 id)))
+      (-> id
+          (db/retrieve request)
+          (a/can-modify? request)
+          (assoc :state "STARTING")
+          (db/edit request))
+      (event-utils/create-event id job-msg (a/default-acl (a/current-authentication request)))
+      (r/map-response job-msg 202 id job-id))
+    (catch Exception e
+      (or (ex-data e) (throw e)))))
+
+
+(defmethod infra-service/post-delete-hook method
+  [service request]
+  (try
+    (let [user-id (:identity (a/current-authentication request))
+          id (:id service)
+          {{job-id     :resource-id
+            job-status :status} :body} (job/create-job id "stop_infrastructure_service_swarm"
+                                                       {:owner {:principal "ADMIN"
+                                                                :type      "ROLE"}
+                                                        :rules [{:principal user-id
+                                                                 :right     "VIEW"
+                                                                 :type      "USER"}]}
+                                                       :priority 50)
+          job-msg (str "stopping " id " with async " job-id)]
+      (when (not= job-status 201)
+        (throw (r/ex-response "unable to create async job to stop infrastructure service swarm" 500 id)))
+      (-> id
+          (db/retrieve request)
+          (a/can-modify? request)
```

Comment:
Again, job to delete the service has been created before the check on whether the user has the rights to do so.

Reply:
fixed